src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -0,0 +1,59 @@
+package se.liu.ida.hefquin.engine.data.impl;
+
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.RDFDataMgr;
+
+import se.liu.ida.hefquin.engine.data.Triple;
+import se.liu.ida.hefquin.engine.data.VocabularyMapping;
+import se.liu.ida.hefquin.engine.query.SPARQLGraphPattern;
+import se.liu.ida.hefquin.engine.query.TriplePattern;
+
+public class VocabularyMappingImpl implements VocabularyMapping{
+	
+	protected final Model vocabularyMapping;
+	
+	public VocabularyMappingImpl() {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+	}
+	
+	
+	// Source: https://jena.apache.org/documentation/io/rdf-input.html
+	public VocabularyMappingImpl(final String rdfFile) {
+		vocabularyMapping = RDFDataMgr.loadModel(rdfFile); //.nt file for N-Triple
+	}
+	
+	public VocabularyMappingImpl(final Set<Triple> tps) {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+		Iterator<Triple> i = tps.iterator();
+		while(i.hasNext()) {
+			Triple tp = i.next();
+			vocabularyMapping.asStatement(tp.asJenaTriple());
```

Comment:
The names of the program variables that you are using here (`tp`and `tps`) seem to indicate that this is about triple patterns rather than triples. To avoid any confusion I suggest to rename them (additionally, mark all local variables as `final` if you are not intending to change them):
```suggestion
	public VocabularyMappingImpl(final Set<Triple> triples) {
		vocabularyMapping = ModelFactory.createDefaultModel();
		final Iterator<Triple> i = triples.iterator();
		while(i.hasNext()) {
			final Triple t = i.next();
			vocabularyMapping.asStatement(t.asJenaTriple());
```